### DISCUSSION TITLE
STM32H743IIT6 Flash Size Error

### DISCUSSION THREAD

**Discussion by Mobin-Mortazavi** [stm32duino/Arduino_Core_STM32#2039] [Variants] [answered]
I wrote a code that easily fits in STM32F103RET6 that has only 512KB of flash , But when I use same arrays in the STM32H743IIT6 it says :
" sketch_jun12b.ino.elf section `.rodata' will not fit in region `FLASH' "
and also :
" arm-none-eabi/bin/ld.exe: region `FLASH' overflowed by 32888 bytes "

when I reduce Array size a little bit , it says that my code only took 6% of the Flash 

My array is a 480*72 uint16 image in the flash , Like this :

const uint16_t image[34560] PROGMEM

**Comment by fpistm** [comment]
Could you be more precise, which target you select? Core version?
Sketch?

**Comment by Mobin-Mortazavi** [comment]
Yes Core Version : 2.5.0
I tested others too... Same problem

I already said my Target was STM32H743IIT6 !

in the arduino : Generic STM32H7 Series => Generic H743IITx

**Reply by fpistm** [reply, marked as answer]
> I already said my Target was STM32H743IIT6 ! 

Please avoid this kind of answer. Yes you told that but sometimes there is several target with the same mcu. That's why I asked to avoid any misunderstanding. Issue #2040 is fixed. Simple fix.

**Reply by Mobin-Mortazavi** [reply]
Thanks my friend , What kind of answer should I avoid ? :)
How is it fixed ? Do you need more explanation?

**Reply by Mobin-Mortazavi** [reply]
The STM32H743IIT6 has 2MB of Flash , Now is it available in your new Fix ? Thank you So much :)
